### ISSUE TITLE
Encoding Detection Issue for PDF Files

### ISSUE THREAD

**Issue by AkshatSoni26** [chardet/chardet#295]
I am encountering an issue with detecting text encoding from PDF files. While the encoding detection works correctly for .txt files, it consistently returns None for PDF files.

Steps to Reproduce:
Attempt to detect encoding from a PDF file using the current method.
Observe that the result is None or not accurate.

Expected Behavior: The encoding detection should correctly identify the encoding of the text extracted from the PDF.

Actual Behavior: The encoding detection returns None, or the result is not accurate.

My Function which I am using to detect encoding:- 

`
  import  chardet

   def detect_file_encoding(file_path):
        with open(file_path, 'rb') as file:
           raw_data = file.read(10000)

       result = chardet.detect(raw_data)

      encoding = result['encoding']
      confidence = result['confidence']

      return encoding, confidence
`

**Comment by slycordinator** [comment]
For pdf files, they'd more or less have to first implement a parser, to convert it to a text file, then pass those bytes into the detector.

**Comment by AkshatSoni26** [comment]
But I didn't want this. 

When converting the content into a text file, the system uses the default encoding unless we explicitly specify one. In my case, I know the encoding of system and don't want the default to be used. Instead, I need a way to detect or determine the encoding from the text of the PDF file itself.

For reference, here’s a link to the official Python documentation: [Python Built-in Functions: open()](https://docs.python.org/3/library/functions.html#open)

**Comment by slycordinator** [comment]
What you are asking for is not possible, currently. 

PDF files often have text that's not stored as text, but instead as the hex bytes in a special format, among others. Some of the text will be plain text. If it uses a different encoding, most of the text won't be encoded as text. 

So, for a chardet-like tool to work with PDFs, it would need to implement a PDF parser to extract the text parts and a way to tell itself how to appropriately convert all of the text to bytes (coverting actual text to bytes and to parse PDF's hex representations into actual bytes, etc).

**Comment by AkshatSoni26** [comment]
okay, Is any other way to do this or any suggestion.

**Comment by dan-blanchard** [comment]
Yeah, PDF parsing is very far out of scope for this project, sorry
